x-pack/plugins/streams_app/public/components/stream_detail_overview/index.tsx

```diff
@@ -0,0 +1,189 @@
+    [share.url.locators]
+  );
+
+  const queries = useMemo(() => {
+    if (!indexPatterns) {
+      return undefined;
+    }
+
+    const baseQuery = `FROM ${indexPatterns.join(', ')}`;
+
+    const bucketSize = Math.round(
+      calculateAuto.atLeast(50, moment.duration(1, 'minute'))!.asSeconds()
+    );
+
+    const histogramQuery = `${baseQuery} | STATS metric = COUNT(*) BY @timestamp = BUCKET(@timestamp, ${bucketSize} seconds)`;
+
+    return {
+      baseQuery,
+      histogramQuery,
+    };
+  }, [indexPatterns]);
+
+  const discoverLink = useMemo(() => {
+    if (!discoverLocator || !queries?.baseQuery) {
+      return undefined;
+    }
+
+    return discoverLocator.getRedirectUrl({
+      query: {
+        esql: queries.baseQuery,
+      },
+    });
+  }, [queries?.baseQuery, discoverLocator]);
+
+  const histogramQueryFetch = useStreamsAppFetch(
+    async ({ signal }) => {
+      if (!queries?.histogramQuery || !indexPatterns) {
+        return undefined;
+      }
+
+      const existingIndices = await dataViews.getExistingIndices(indexPatterns);
+
+      if (existingIndices.length === 0) {
+        return undefined;
+      }
+
+      return streamsRepositoryClient.fetch('POST /internal/streams/esql', {
+        params: {
+          body: {
+            operationName: 'get_histogram_for_stream',
+            query: queries.histogramQuery,
+            start,
+            end,
+          },
+        },
+        signal,
+      });
+    },
+    [indexPatterns, dataViews, streamsRepositoryClient, queries?.histogramQuery, start, end]
+  );
+
+  const dataViewsFetch = useAbortableAsync(() => {
+    return dataViews
+      .create(
+        {
+          title: dataStream,
+          timeFieldName: '@timestamp',
+        },
+        false, // skip fetch fields
+        true // display errors
+      )
+      .then((response) => {
+        return [response];
+      });
+  }, [dataViews, dataStream]);
+
+  const fetchedDataViews = useMemo(() => dataViewsFetch.value ?? [], [dataViewsFetch.value]);
+
+  return (
+    <>
+      <EuiFlexGroup direction="column">
+        <EuiFlexGroup direction="row" gutterSize="s">
+          <EuiFlexItem grow>
+            <StreamsAppSearchBar
+              onQuerySubmit={({ dateRange }, isUpdate) => {
+                if (!isUpdate) {
+                  histogramQueryFetch.refresh();
+                  return;
+                }
+
+                if (dateRange) {
+                  setTimeRange({ from: dateRange.from, to: dateRange?.to, mode: dateRange.mode });
+                }
+              }}
+              onRefresh={() => {
+                histogramQueryFetch.refresh();
+              }}
+              placeholder={i18n.translate(
+                'xpack.streams.entityDetailOverview.searchBarPlaceholder',
+                {
+                  defaultMessage: 'Filter data by using KQL',
+                }
+              )}
+              dataViews={fetchedDataViews}
```

Comment:
looks like we don't actually need this as we only use the time picker and refresh button